src/arch/riscv/tlb.cc

```diff
@@ -366,6 +383,15 @@ TLB::translate(const RequestPtr &req, ThreadContext *tc,
 
         if (!delayed && fault == NoFault) {
             pma->check(req);
+            // do pmp checks if any condition is met
+
+            if (pmp->shouldCheckPMP(pmode, mode, tc)){
```

Comment:
Can you hide this check in `pmpCheck`? In that function, can you first check if it's enabled/should be checked and return noFault in that case?

Reply:
done